src/main/java/net/snowflake/ingest/streaming/internal/StreamingIngestUtils.java

```diff
@@ -0,0 +1,114 @@
+package net.snowflake.ingest.streaming.internal;
+
+import static net.snowflake.ingest.utils.Constants.MAX_API_RETRY;
+import static net.snowflake.ingest.utils.Constants.RESPONSE_ERR_GENERAL_EXCEPTION_RETRY_REQUEST;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import java.io.IOException;
+import java.util.Map;
+import java.util.function.Function;
+import net.snowflake.ingest.connection.IngestResponseException;
+import net.snowflake.ingest.connection.RequestBuilder;
+import net.snowflake.ingest.connection.ServiceResponseHandler;
+import net.snowflake.ingest.utils.ErrorCode;
+import net.snowflake.ingest.utils.SFException;
+import org.apache.http.client.HttpClient;
+
+public class StreamingIngestUtils {
+
+  private static class DefaultStatusGetter<T extends StreamingIngestResponse>
+      implements Function<T, Long> {
+    public DefaultStatusGetter() {}
+
+    public Long apply(T input) {
+      return input.getStatusCode();
+    }
+  }
+
+  private static final DefaultStatusGetter defaultStatusGetter = new DefaultStatusGetter();
+
+  private static final ObjectMapper objectMapper = new ObjectMapper();
+
+  static <T extends StreamingIngestResponse> T executeWithRetries(
```

Comment:
Why we have three flavors of `executeWithRetries`, do you think we could simplify it?

Reply:
Two of the methods are because we pass both Map and String payloads, to this function so it was easy to overload the method to handle both inputs.  The third version is because client configure response is not well represented by a pojo in the client so I have a method that can handle getting the status code from a Map.  Note that there is no duplicated logic here, just different signatures for the method that call eachother

Reply:
I think we should create a pojo instead of adding a third function here?